src/libraries/System.Linq/src/System/Linq/AnyAll.cs

```diff
@@ -10,36 +9,13 @@ public static partial class Enumerable
     {
         public static bool Any<TSource>(this IEnumerable<TSource> source)
         {
-            if (source == null)
-            {
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.source);
-            }
-
-            if (source is ICollection<TSource> collectionoft)
-            {
-                return collectionoft.Count != 0;
-            }
-            else if (source is IIListProvider<TSource> listProv)
-            {
-                // Note that this check differs from the corresponding check in
-                // Count (whereas otherwise this method parallels it).  If the count
-                // can't be retrieved cheaply, that likely means we'd need to iterate
-                // through the entire sequence in order to get the count, and in that
-                // case, we'll generally be better off falling through to the logic
-                // below that only enumerates at most a single element.
-                int count = listProv.GetCount(onlyIfCheap: true);
-                if (count >= 0)
-                {
-                    return count != 0;
-                }
-            }
-            else if (source is ICollection collection)
-            {
-                return collection.Count != 0;
-            }
+            return
+                TryGetNonEnumeratedCount(source, out int count) ? count != 0 :
```

Comment:
@stephentoub do we care that any `ArgumentNullException` will now be thrown from `TryGetNonEnumeratedCount`, resulting in less clean stacktraces?

Reply:
In a debug build Any will still be in the call stack.  I think it's fine.